### PR TITLE
Task-99-updated app to useHistory

### PR DESCRIPTION
**Pull request by TelescopeThieves**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)

## Questions and concerns 

I updated the app to make use of the useHistory react hook and abstracted the logout function from the footer so that it's passed down as a prop. 

Unfortunately, as far as moving beyond the need to do a window refresh on logout, I was unsuccessful. If we want to have the feed re-render when someone clicks the logout button, useHistory is not the solution.

### CODE REVIEW

File: packages/client/src/components/TextLink.js [hunk 1]
```diff
@@ -11,9 +12,7 @@ const TextLink = (props) => {
       padding={props.padding}
       onClick={props.onClick}
     >
-      <Link to={props.link}>
-        <span>{props.text}</span>
-      </Link>
+      <span onClick={() => history.push(props.onClickGoTo)}>{props.text}</span>
```
Comment (indifferentghost) [inline packages/client/src/components/TextLink.js:15]: I actually think using Link works better here.

**Review by indifferentghost — CHANGES_REQUESTED** [review]
useHistory works better in _most_ places where you'd be using a redirect based on toggling. But default navigation should use links if applicable.

File: packages/client/src/components/TextLink.js (hunk 1 above)
Comment (TelescopeThieves) [reply packages/client/src/components/TextLink.js:15]: cool, I'll revert this and do another commit in a bit!

**Comment by TelescopeThieves** [comment]
Going to re-do this pull request to avoid conflicts

**Pull request closed without merging**